examples/spin-timer/deps/spin/variables.wit

```diff
@@ -0,0 +1,13 @@
+package fermyon:spin@2.0.0
+
+interface variables {
+    // Get an application variable value for the current component.
+    // The name must match one defined in in the component manifest.
+    get: func(name: string) -> result<string, error>
+
+    variant error {
+        invalid-name(string),
+        provider(string),
+        other(string),
+    }
```

Comment:
Do we want to have a `not-found` error for when the requested variable is not found? Right now this gets encoded as a `provider` error which doesn't seem right.

Reply:
I left it out because I didn't want to imply that `not-found` was a "normal" condition (like the KV v1 equivalent); not being able to resolve a variable is a programming or operational mistake.

Reply:
It's strange that this error is labeled as a provider error when it's fairly likely *not* a provider error. I think the most likely cause will be users thinking there is a variable name `foo` and there not actually being one. 

It's not a huge deal that this gets rendered as `provider("no variable named 'foo'")` - the error is clear enough - it's just a bit strange is all. 

I personally don't really see potential confusion from having another error case, but I guess I'm fine letting this go if everyone disagrees with me.